mlflow/protos/service.proto

```diff
@@ -3069,3 +3109,77 @@ message Scorer {
   // The creation time of the scorer version (in milliseconds since epoch).
   optional int64 creation_time = 5;
 }
+
+// =============================================================================
+// Prompt Optimization Messages
+// =============================================================================
+
+// Request to start a prompt optimization job.
+message OptimizePrompts {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // ID of the training dataset.
+  optional string train_dataset_id = 1 [(validate_required) = true];
+  
+  // ID of the evaluation dataset.
+  optional string eval_dataset_id = 2 [(validate_required) = true];
```

Comment:
I think we can make it optional and split train dataset if eval dataset is empty